modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala

```diff
@@ -6,24 +6,19 @@ import java.nio.file.Path
 import scala.language.higherKinds
 import scala.reflect.ClassTag
 
-import cats.data.NonEmptyList
-import cats.effect.Sync
+import cats.effect.{ Blocker, ContextShift, Sync }
 import cats.implicits._
-import com.typesafe.config.{ ConfigRenderOptions, Config => TypesafeConfig }
+import com.typesafe.config.ConfigRenderOptions
 import pureconfig._
 import pureconfig.error.ConfigReaderException
 
 package object catseffect {
 
-  @deprecated("Root will be treated as the default namespace", "0.12.0")
-  val defaultNameSpace = ""
+  private[catseffect] type ConfReader[A] = Derivation[ConfigReader[A]]
+  private[catseffect] type ConfWriter[A] = Derivation[ConfigWriter[A]]
 
-  def loadF[F[_], A](cs: ConfigSource)(implicit F: Sync[F], reader: Derivation[ConfigReader[A]], ct: ClassTag[A]): F[A] = {
-    val delayedLoad = F.delay {
-      cs.load[A].leftMap[Throwable](ConfigReaderException[A])
-    }
-    delayedLoad.rethrow
-  }
+  def loadF[F[_]: Sync: ContextShift, A: ConfReader: ClassTag](cs: ConfigSource, blocker: Blocker): F[A] =
+    blocker.delay(cs.load[A].leftMap[Throwable](ConfigReaderException[A])).rethrow
```

Comment:
Note that reading `A` is still done in the blocking context. Ideally, this would be something like
```scala
EitherT(blocker.delay[F] { cs.cursor() })
  // back on `ContextShift`
  .subflatMap(_.to[A])
  .leftMap(ConfigReaderException[A])
  .rethrowT
```
Although, this would potentially be confusing if something overrides `load` in their `ConfigSource` implementation. @ruippeixotog, WDYT?

Reply:
Yeah, I thought about the same. Ideally only the blocking part should be done inside blocker, but I wasn't sure which method should have I called instead.

Maybe, `load` and `loadOrThrow` could / should be final?

Reply:
I agree with @nigredo-tori suggestion and with making `load` and `loadOrThrow` final, as they were never intended to be overridden in the first place.